### PR TITLE
[NEAT-390] ⛓Conversion methods as Rule Transformations Part 2

### PR DESCRIPTION
**Pull request by doctrino**
Replaces all conversion methods with Rule Transformers.

**Notice**: I set a `## TBD` heading in the changelog. This is to try to gather up more changes before making another release.

### CODE REVIEW

File: cognite/neat/graph/extractors/_mock_graph_generator.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 from cognite.neat.rules.models.data_types import DataType
 from cognite.neat.rules.models.entities import ClassEntity, EntityTypes
 from cognite.neat.rules.models.information import InformationProperty
+from cognite.neat.rules.transformers import DMSToInformation
```
Comment (nikokaoja) [inline cognite/neat/graph/extractors/_mock_graph_generator.py:20]: I am always a bit puzzled what to do with all caps acronyms when defining a class.

File: cognite/neat/graph/extractors/_mock_graph_generator.py [hunk 2]
```diff
@@ -42,7 +43,7 @@ def __init__(
         allow_isolated_classes: bool = True,
     ):
         if isinstance(rules, DMSRules):
-            self.rules = rules.as_information_rules()
+            self.rules = DMSToInformation().transform(rules)
```
Comment (nikokaoja) [inline cognite/neat/graph/extractors/_mock_graph_generator.py:46]: nice :) , clear

File: cognite/neat/graph/extractors/_mock_graph_generator.py [hunk 3]
```diff
@@ -17,6 +17,7 @@
 from cognite.neat.rules.models.data_types import DataType
```
Comment (nikokaoja) [inline cognite/neat/graph/extractors/_mock_graph_generator.py:17]: Do we have those fancy repl_html you made for graph also here?

File: cognite/neat/rules/models/asset/_rules.py [hunk 4]
```diff
@@ -22,7 +21,7 @@
 )
 
 if TYPE_CHECKING:
```
Comment (nikokaoja) [inline cognite/neat/rules/models/asset/_rules.py:23]: do you need this ? if not used ?

**Review by nikokaoja — APPROVED** [review]

File: cognite/neat/rules/models/asset/_rules.py (hunk 4 above)
Comment (doctrino) [reply cognite/neat/rules/models/asset/_rules.py:23]: This is removed in later PRs.

**Pull request merged**